### PR TITLE
Dependency updates v2024.2

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### Why is this the best possible solution? Were any other approaches considered?
As usual, I've updated the dependencies for the next release. The only thing that looks odd is https://github.com/getodk/collect/pull/5974/commits/eec50c199f35f7974f1d7d4b7588afba75eea3aa
It looks like the newer robolectric version does a better job and it revealed that our tests had some mistakes. The main issue was that in some tests we tried to verify dates that were out of range. For example, in the Coptic date widget, we support years 1617-1817 but the test tried to use 2020. I've reworked those tests so that they all test 2020 but in our Gregorian calendar.
https://github.com/getodk/collect/commit/eec50c199f35f7974f1d7d4b7588afba75eea3aa#diff-68a252dfcc4fa9e25ba79156f878ea52f1926dfa3420e03412eb03c419baac4fL89

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I think we can merge this without testing like we usually do if there are no any specific changes.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
+root = true
+
+[*.{kt,kts}]
+ktlint_standard_no-blank-lines-in-chained-method-calls = disabled
```
Comment (grzesiek2010) [inline .editorconfig:4]: There is a couple of new rules I had to disable. I have fixed some of them in commits: https://github.com/getodk/collect/pull/5974/commits/9b7e6a47472f3f14ee76d33ac09fcbc72c0053b2 https://github.com/getodk/collect/pull/5974/commits/1ed546d4e2225b48925c017340943dbfb9dbabfd https://github.com/getodk/collect/pull/5974/commits/64c21076d055287a126f903b52fba9e52509fe82 https://github.com/getodk/collect/pull/5974/commits/9c57ee472ed2f140b53b872a20b40bda0be63567 https://github.com/getodk/collect/pull/5974/commits/0bd66c2657df2a0a9ae5f7f2c4d0b1c53c22d3c3
the others would either require updating a lot of classes or they do not make that much sense so it's something for separate pull requests.

File: docs/STATE.md [hunk 2]
```diff
@@ -45,4 +45,3 @@ and update this document as the code evolves.
 * Replacing async work such as `AsyncTask` with `LiveData` + `Scheduler` abstraction
 * Gradually removing use of `CursorLoader` (all remaining uses are in `CursorLoaderFactory`)
 * Using AndroidX Test in new local tests and migrating other local tests as we touch them (from classic Robolectric)
-* Replacing uses of Google Guava with similar helpers in Kotlin Standard Library
```
Comment (seadowg) [inline docs/STATE.md:48]: 🎉🥳

File: robolectric-deps.properties [hunk 3]
```diff
@@ -7,3 +7,4 @@ org.robolectric\:android-all-instrumented\:12-robolectric-7732740-i3=../../../..
 org.robolectric\:android-all-instrumented\:12-robolectric-7732740-i4=../../../../../../robolectric-deps/android-all-instrumented-12-robolectric-7732740-i4.jar
 org.robolectric\:android-all-instrumented\:12.1-robolectric-8229987-i4=../../../../../../robolectric-deps/android-all-instrumented-12.1-robolectric-8229987-i4.jar
 org.robolectric\:android-all-instrumented\:13-robolectric-9030017-i4=../../../../../../robolectric-deps/android-all-instrumented-13-robolectric-9030017-i4.jar
+org.robolectric\:android-all-instrumented\:14-robolectric-10818077-i4=../../../../../../robolectric-deps/android-all-instrumented-14-robolectric-10818077-i4.jar
```
Comment (seadowg) [inline robolectric-deps.properties:10]: I think we can probably ditch the 11 and 12 SDKs now?

**Review by seadowg — CHANGES_REQUESTED** [review]
Some great changes and clean up here! It looks like the new ktlint rules don't like the `mapbox` though.

File: robolectric-deps.properties (hunk 3 above)
Comment (grzesiek2010) [reply robolectric-deps.properties:10]: It's still needed. The build fails if I remove it.

**Comment by grzesiek2010** [comment]
> It looks like the new ktlint rules don't like the mapbox though.

Fixed.

**Review by seadowg — APPROVED** [review]

**Comment by seadowg** [comment]
@grzesiek2010 looks like this has conflicts to fix before merge

**Pull request merged**